### PR TITLE
Casos examples

### PR DESCRIPTION
**Pull request by JOlucak**
Demos-reachability folder renamed to casos-examples. casos-examples is up-to-date with the main (today).
In the demo folder, the examples are moved to a separate folder to have a cleaner appearence. 
Examples that are not working (compared to sosopt) are added to a separate folder.

### CODE REVIEW

File: demo/roaest.m
Comment (tcunis) [inline demo/roaest.m:1]: This file seems to be a duplicate of `02_ROA/roaest`.

File: demo/roaest_gtm.m
Comment (tcunis) [inline demo/roaest_gtm.m:1]: This file seems to be a duplicate of `02_ROA/roaest_gtm`.

**Review by tcunis — CHANGES_REQUESTED** [review]
I have some issues with this PR in its current form. Please work with me to get it improved:

1. This PR makes unnecessary changes to core files and overlaps with #42. I think we should either (I have a strong preference for the third option):
   - set up a new branch from the current main and only add the examples file, 
   - hard-remove unnecessary changes from the history and push as new branch,
   - do an interactive rebase onto the current main where unnecessary changes are removed, then push as new branch.

2. Moving forward, we are going to have separate top-level folders for demos and examples, as established by #37, with the following distinction:
   - `demo`: very simple files that demonstrate how to use the toolbox. These include `polyopt`, `projsos`, `localstab` (previously `roaest`), and `roaest` (previously `roaest2`). If there are any changes to the demo files, please explain why these were necessary.
   - `examples`: more elaborate scripts, in particular specific applications and/or examples from the literature and (future) publications, that show off the results of using the toolbox.

3. I like the subfolder structure in the examples folder, but I think the files within those subfolders could be structure better. Is it clear for the user which scripts to use and how? I'd suggest:
   - introduce a naming convention for example scripts and stick to it,
   - add a very brief `readme.md` file to each subfolder that explains the general ideas and which scripts correspond to which examples,
   - move auxiliary files (such as `mat` files) to `private` folders within the subfolders.
4. A brief explanation why (or which part of) the "not working" examples do not work would be helpful.

**Comment by JOlucak** [comment]
> I have some issues with this PR in its current form. Please work with me to get it improved:
> 
>     1. This PR makes unnecessary changes to core files and overlaps with [New polynomial operations #42](https://github.com/iFR-ACSO/casos/pull/42). I think we should either (I have a strong preference for the third option):
>        
>        * set up a new branch from the current main and only add the examples file,
>        * hard-remove unnecessary changes from the history and push as new branch,
>        * do an interactive rebase onto the current main where unnecessary changes are removed, then push as new branch.
> 
>     2. Moving forward, we are going to have separate top-level folders for demos and examples, as established by [Cones supported by solvers #37](https://github.com/iFR-ACSO/casos/pull/37), with the following distinction:
>        
>        * `demo`: very simple files that demonstrate how to use the toolbox. These include `polyopt`, `projsos`, `localstab` (previously `roaest`), and `roaest` (previously `roaest2`). If there are any changes to the demo files, please explain why these were necessary.
>        * `examples`: more elaborate scripts, in particular specific applications and/or examples from the literature and (future) publications, that show off the results of using the toolbox.
> 
>     3. I like the subfolder structure in the examples folder, but I think the files within those subfolders could be structure better. Is it clear for the user which scripts to use and how? I'd suggest:
>        
>        * introduce a naming convention for example scripts and stick to it,
>        * add a very brief `readme.md` file to each subfolder that explains the general ideas and which scripts correspond to which examples,
>        * move auxiliary files (such as `mat` files) to `private` folders within the subfolders.
> 
>     4. A brief explanation why (or which part of) the "not working" examples do not work would be helpful.

Fair enough. I think three is the best option. I will work on that. I will also reduce the examples to a very minimum for now.

File: demo/examples/09_notWorking/notWorking/car2d.m
[the remark this answers is not in the document]
Comment (JOlucak) [reply demo/examples/09_notWorking/notWorking/car2d.m:1]: This was a mistake. I will change that

**Pull request closed without merging**